aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -38,20 +38,41 @@ class PwBaseWorkChain(BaseRestartWorkChain):
 
     @classmethod
     def define(cls, spec):
+        # yapf: disable
         super(PwBaseWorkChain, cls).define(spec)
-        spec.input('code', valid_type=orm.Code)
-        spec.input('structure', valid_type=orm.StructureData)
-        spec.input('kpoints', valid_type=orm.KpointsData, required=False)
-        spec.input('kpoints_distance', valid_type=orm.Float, required=False)
-        spec.input('kpoints_force_parity', valid_type=orm.Bool, required=False)
-        spec.input('parameters', valid_type=orm.Dict)
-        spec.input_namespace('pseudos', required=False, dynamic=True)
-        spec.input('pseudo_family', valid_type=orm.Str, required=False)
-        spec.input('parent_folder', valid_type=orm.RemoteData, required=False)
-        spec.input('vdw_table', valid_type=orm.SinglefileData, required=False)
-        spec.input('settings', valid_type=orm.Dict, required=False)
-        spec.input('options', valid_type=orm.Dict, required=False)
-        spec.input('automatic_parallelization', valid_type=orm.Dict, required=False)
+        spec.input('code', valid_type=orm.Code,
+            help='The code to use for the calculation which is configured to a Quantum ESPRESSO `pw.x` executable.')
+        spec.input('structure', valid_type=orm.StructureData,
+            help='The input structure.')
+        spec.input('kpoints', valid_type=orm.KpointsData, required=False,
+            help='An explicit k-points list or mesh. Either this or `kpoints_distance` has to be provided.')
+        spec.input('kpoints_distance', valid_type=orm.Float, required=False,
+            help='The desired distance in 1/Å between k-points in reciprocal space. The explicit k-points will be'
```

Comment:
Is this the minimum or maximum distance betweek k-points? I always forget...

Reply:
minimum, I added it to the message